admin/app/controllers/cache/PageDecacheController.scala

```diff
@@ -9,31 +9,62 @@ import controllers.admin.AuthActions
 import model.NoCache
 import play.api.libs.ws.WSClient
 import play.api.mvc.Security.AuthenticatedRequest
-import play.api.mvc.{Action, AnyContent, Controller}
+import play.api.mvc._
 import purge.CdnPurge
+import tools.LoadBalancer
 
+import scala.concurrent.Future
 import scala.concurrent.Future.successful
 
+case class PrePurgeTestResult(url: String, passed: Boolean)
+
 class PageDecacheController(wsClient: WSClient) extends Controller with Logging with ExecutionContexts {
 
-  def renderPageDecacheForm() = Action { implicit request =>
-    NoCache(Ok(views.html.cache.pageDecacheForm()))
+  def renderPageDecache(url: Option[String] = None) = Action.async { implicit request =>
+    url match {
+      case Some(s) => renderPrePurgeTestResult(s)
+      case None => Future(NoCache(Ok(views.html.cache.pageDecache())))
+    }
+  }
+
+  private def renderPrePurgeTestResult(purgeUrl: String)(implicit request: Request[AnyContent]) = {
+    getRouterUrl(purgeUrl).map { routerUrl =>
+      wsClient.url(routerUrl)
+        .withRequestTimeout(2000)
+        .withVirtualHost("www.theguardian.com")
+        .get().map {
+          response => PrePurgeTestResult(purgeUrl, response.status == 200)
+
+      } recoverWith {
+        case t: Throwable => successful(PrePurgeTestResult(purgeUrl, passed = false))
+      } map { result =>
+        NoCache(Ok(views.html.cache.pageDecache(Some(result))))
+      }
```

Comment:
Formatting: I prefer not to mix together dot and space notations.
Also I also prefer to have one method call per line.
What do you think?